ext/spl/spl_observer.c

```diff
@@ -266,35 +265,35 @@ static zend_object *spl_object_storage_new_ex(zend_class_entry *class_type, zend
 
 	zend_hash_init(&intern->storage, 0, NULL, spl_object_storage_dtor, 0);
 
-	while (parent) {
-		if (parent == spl_ce_SplObjectStorage) {
-			/* Possible optimization: Cache these results with a map from class entry to IS_NULL/IS_PTR.
-			 * Or maybe just a single item with the result for the most recently loaded subclass. */
-			if (class_type != spl_ce_SplObjectStorage) {
-				zend_function *get_hash = zend_hash_str_find_ptr(&class_type->function_table, "gethash", sizeof("gethash") - 1);
-				if (get_hash->common.scope != spl_ce_SplObjectStorage) {
-					intern->fptr_get_hash = get_hash;
-				}
-				if (intern->fptr_get_hash != NULL ||
-					SPL_OBJECT_STORAGE_CLASS_HAS_OVERRIDE(class_type, zf_offsetget) ||
-					SPL_OBJECT_STORAGE_CLASS_HAS_OVERRIDE(class_type, zf_offsetexists)) {
-					intern->flags |= SOS_OVERRIDDEN_READ_DIMENSION;
-				}
-
-				if (intern->fptr_get_hash != NULL ||
-					SPL_OBJECT_STORAGE_CLASS_HAS_OVERRIDE(class_type, zf_offsetset)) {
-					intern->flags |= SOS_OVERRIDDEN_WRITE_DIMENSION;
-				}
-
-				if (intern->fptr_get_hash != NULL ||
-					SPL_OBJECT_STORAGE_CLASS_HAS_OVERRIDE(class_type, zf_offsetunset)) {
-					intern->flags |= SOS_OVERRIDDEN_UNSET_DIMENSION;
-				}
-			}
-			break;
+	/* Child classes may overload internal access methods */
+	if (class_type != spl_ce_SplObjectStorage) {
```

Comment:
This breaks, and causes test failures, because MultipleIterator _also_ uses this function: 
`spl_ce_MultipleIterator->create_object = spl_SplObjectStorage_new;`...
So you have to change the condition a bit...